docs/Troubleshooting.md

```diff
@@ -1,137 +1,123 @@
 # Troubleshooting
 
-*  ```
-   perf_event mmap failed: Operation not permitted
-   ```
-   Profiler allocates 8kB perf_event buffer for each thread of the target process.
-   Make sure `/proc/sys/kernel/perf_event_mlock_kb` value is large enough
-   (more than `8 * threads`) when running under unprivileged user. Otherwise, the above message
-   will be printed, and no native stack traces will be collected.
-
-*  ```
-   Failed to change credentials to match the target process: Operation not permitted
-   ```
-   Due to limitation of HotSpot Dynamic Attach mechanism, the profiler must be run
-   by exactly the same user (and group) as the owner of target JVM process.
-   If profiler is run by a different user, it will try to automatically change
-   current user and group. This will likely succeed for `root`, but not for
-   other users, resulting in the above error.
-
-
-*  ```
-   Could not start attach mechanism: No such file or directory
-   ```
-   The profiler cannot establish communication with the target JVM through UNIX domain socket.
-   Usually this happens in one of the following cases:
-     1. Attach socket `/tmp/.java_pidNNN` has been deleted. It is a common
-     practice to clean `/tmp` automatically with some scheduled script.
-     Configure the cleanup software to exclude `.java_pid*` files from deletion.  
-     How to check: run `lsof -p PID | grep java_pid`  
-     If it lists a socket file, but the file does not exist, then this is exactly
+## perf_event mmap failed: Operation not permitted
+
+Profiler allocates 8kB perf_event buffer for each thread of the target process.
+Make sure `/proc/sys/kernel/perf_event_mlock_kb` value is large enough
+(more than `8 * threads`) when running under unprivileged user. Otherwise, the above message
+will be printed, and no native stack traces will be collected.
+
+## Failed to change credentials to match the target process: Operation not permitted
```

Comment:
Error messages as titles look... strange. Especially when they are that large. I'd make them at least two levels smaller.

Reply:
I think it is fair to have error messages as title, in the Troubleshooting page. Made them smaller.